MAINTAINING.md

```diff
@@ -96,3 +96,401 @@ write effective API documentation, such as setting up the RST files and docstrin
 If changes you are making affect non-API reference content in https://quantum.cloud.ibm.com/docs
 you can open an issue (or better yet a PR) to update the relevant page in https://github.com/Qiskit/documentation.
 You can also use this repo to suggest or contribute brand new content beyond updates to the API reference.
+
+
+## The release manager role
+
+The processes around tracking that the correct code goes into a release in a timely manner, and releasing the package when it is time is the responsibility of the **release manager**.
+
+Each release is tracked through a milestone that aggregates all associated issues and PRs.
+This milestone should be continuously groomed throughout the release cycle. Additionally, at the release manager's discretion, a GitHub project board may be used to help prioritize tasks and monitor overall progress.
+During the release cycle, the release manager is responsible for:
+
+- Tracking the status of the remaining issues.
+
+- Determining when issues/PRs need to be pushed to a later release, and which are high priority and the release should block on.
+  Discuss with the team; you don't need to make all the decisions, you're just responsible for making sure
+  the decisions are made.
+
+- Making sure there are sufficient coders and reviewers assigned to each PR, and checking up on them to make sure they're progressing.
+
+- If this release is a release candidate for a major or minor release, there is a feature freeze starting two weeks before release day.
```

Comment:
As this bullet is phrased, it seems to be out of place since it's about some detailed aspect of the iteration process rather than the general role of the release manager. I think this should be moved to the Check the milestone state section below, what do you say?